hack/generator/pkg/codegen/testdata/EmbeddedTypes/Embedded_type_simple_resource.golden

```diff
@@ -43,7 +43,7 @@ type FakeResource_SpecArm struct {
 	ApiVersion       FakeResourceSpecApiVersion `json:"apiVersion"`
 	Color            *FakeResourceSpecColor     `json:"color,omitempty"`
 	Foo              FooArm                     `json:"foo"`
-	Name             string                     `json:"name"`
+	Name             string                     `json:"name,omitempty"`
```

Comment:
This seems weird and makes me think that the right value for `jsonOmitIfEmpty(PrimitiveType)` is false. Why shouldn't it be?

Reply:
Partly this is because _whether-we-serialize_ is independent from _this-property-is-mandatory_.

Also, while the primary consumer of these files is another process, they are visible to end users; leaving out properties that have been left blank reduces the amount of dross that end users need to wade through. This is particularly relevant when we're reporting on errors, as many of those have extra fields that don't provide any value when blank.